idrstream/idr_stream.py

```diff
@@ -191,18 +238,31 @@ def init_segmentor(self, model_specs: dict):
         self.segmentor = segment.CellPoseSegmentor(model_specs)
         self.logger.info("CellPose segmentor initialized")
 
-    def prepare_batch(self, batch_metadata: pd.DataFrame):
+    def init_cp_metadata(self, data_to_process_tsv: pathlib.Path):
+        """initialize cellprofiler metadata compiler
+
+        Parameters
+        ----------
+        data_to_process : pd.DataFrame
+            metadata for CellProfiler to process
         """
-        download, preprocess, and segment data from a batch
+        self.cp_metadata = cellprofiler_metadata.convert_tsv_to_csv(data_to_process_tsv)
```

Comment:
It looks like `cellprofiler_metadata.convert_tsv_to_csv()` does not return anything, but you use it set a variable. Did you mean to return a pandas dataframe in that function?

Reply:
@roshankern 

It does return something, it creates the `data_to_process.csv` file in the `/example_files` folder so that it can then be copied into the `/tmp/CP_project/inputs/images' folder.

Reply:
I meant the `cellprofiler_metadata.convert_tsv_to_csv()` function does not have the `return` keyword in it, and thus does not return a value when it is called. `self.cp_metadata` must be null after this line if nothing is returned so it might be more appropriate to change this line to just `cellprofiler_metadata.convert_tsv_to_csv(data_to_process_tsv)`

Reply:
(also I don't see `self.cp_metadata` used anywhere else so it must not be a necessary variable for the stream to function)